scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -834,20 +834,18 @@ def check_zephyr_version(self):
                 if _version:
                     self.version = _version
                     logger.info(f"Zephyr version: {self.version}")
-                else:
-                    self.version = "Unknown"
-                    logger.error("Could not determine version")
         except OSError:
-            logger.info("Cannot read zephyr version.")
```

Comment:
From what I know, `Except: pass` constructs are to be avoided, since they can hide a real issue. I believe a logging should stay, at least at a debug level. Maybe like this?
```
except OSError as e:
    logger.info(f"Cannot read zephyr version: {str(e)}")
```

Reply:
To be honest, the likeliness of an `OSError` on `subprocess.run` is so low that you'd probably never be hitting it in the first place, and I was of two minds of not catching it at all, and just let it bubble up — it is not caught on the following call for `git show` so I'm not quite sure why this is treated differently.

Though we should really use `logging.exception("Failure when reading Zephyr version")` to print the stack trace if we want the log.

Which one do you prefer?

Reply:
I opt for `logger.exception`. Thanks